src/net/sourceforge/kolmafia/textui/javascript/JavascriptRuntime.java

```diff
@@ -112,7 +112,7 @@ public static List<net.sourceforge.kolmafia.textui.parsetree.Function> getFuncti
       for (VariableReference variableReference : libraryFunction.getVariableReferences()) {
         allTypes.add(variableReference.getType());
       }
-      if (allTypes.contains(DataTypes.MATCHER_TYPE)) {
+      if (allTypes.contains(DataTypes.MATCHER_TYPE) || allTypes.contains(DataTypes.RNG_TYPE)) {
```

Comment:
I think we want them to specifically come through to JavaScript - the reason we ban matchers is because regular expressions are handled by JavaScript separately.

Reply:
I figured TypeScript users can use https://www.npmjs.com/package/kol-rng instead, unless there's some reason it's preferable for them to use the Mafia version?

Reply:
I can't figure out how to make JavaScript work with this.

I get
> Script exception: ASH function returned native Java object.

and uncommenting the error means the object is returned as a string instead. I tried wrapping it (like you wrapped `sessionStorage`) to get a `NativeJavaObject` explicitly, but it had the same problem.